### DISCUSSION TITLE
Refinement of survey responses - features

### DISCUSSION THREAD

**Discussion by bertfrees** [daisy/pipeline#712] [General]
We are currently in the process of analysing your responses to the [Pipeline usage survey](https://github.com/daisy/pipeline/discussions/682) held earlier this year. Some of the feedback requires some more elaboration or discussion. Below are some responses in the topic "features":

---
Some organisations requested new TTS related features:

- **Q**: What do you think are the 3 greatest weaknesses of Pipeline?
  > More option to generate TTS DTBooks in Pipeline 2, for example: smil type o Mp3 rate
- **Q**: Please describe any functionality you would like to see in Pipeline in future
  > A Text-to-speech connector for the acapela-group TTS API
- **Q**: Please describe any functionality you would like to see in Pipeline in future
  > Support for TTS in cloud (Google etc.).

"Better TTS support" was also brought up as a issue by another organisation:

- **Q**: What do you think are the 3 greatest weaknesses of Pipeline?
  > - Better TTS support, simpler setup
  > - Better support for multiple languages

Conversion from DOCX to DTBook has been a request for some time and still is (was mentioned by two organisations):

- **Q**: Please describe any functionality you would like to see in Pipeline in future
  > - Conversion DOCX to DTBook
  > - A word to DTBook XML convertor (outside or linked to Save As DAISY word plugin)

Other responses include:

- **Q**: Please describe any functionality you would like to see in Pipeline in future
  > An easy way to produce braille and epub from HTML5 files
- **Q**: Please describe any functionality you would like to see in Pipeline in future
  > - Customisable braille and large print formatter
  > - The ability to add multiple jobs from subfolders.
---

I would like to know:

- What exactly do you mean by "smil type"?
- With "mp3 rate", do you mean bit rate?
- We would like some more information about the "acapela-group TTS API".
- Regarding TTS in the cloud: Is the current support for Google TTS useful for you? Will the upcoming [support for Microsoft neural TTS](https://github.com/daisy/pipeline/discussions/691) be useful? Do you have other suggestions?
- Do you have suggestions to improve the TTS support? Do you have some specific issues? We're currently trying to simplify the setup of TTS and could use some suggestions.
- Does "better support for multiple languages" refer to TTS as well?
- We would like to get some specific issues and suggestions for the existing "HTML to braille" script in Pipeline 2. Can you define "easy"? Can you define "customisable"? We want to simplify the configuration of this script, and any suggestions are welcome.
- Similarly, we would like to get some specific issues and suggestions for the existing "HTML to EPUB 3" script in Pipeline 2.
- Regarding the ability to add multiple jobs from subfolders: do you already have an idea of how this would work exactly? Would this be a graphical user interface feature or a more general one?

**Comment by torchtrust** [comment]
Regarding the multiple jobs from subfolders: A number of times I have wanted to validate a whole swathe of DAISY titles. It would be easier to use the UI to select a folder that contains all the subfolders I want validating and report any ncc.html that comes back with a fail on the validator. I have done this using the command line, but requiring both the source and then the validation report folder was a bit of a pain.
HTH
Paul

**Reply by bertfrees** [reply]
I'm thinking this would make most sense as a UI feature. But coming up with a good interface is tricky. There are some things to take into account. The results and messages of the different conversions need to stay separated, so we need to launch one job per book. We probably don't want to open a job tab for each new job, as there might be a lot of jobs. This means the jobs should start in the background, and we need a notification system to bring them to the front.

(Another approach is to adapt certain scripts to take a folder containing multiple books as input and then processing the books one by one. But this sounds a bit too limiting, and a generic solution would be nicer.)
